lib/engine/game/base.rb

```diff
@@ -560,6 +559,12 @@ def init_hexes(companies, corporations)
           end
         end
 
+        reservations = Hash.new { |k, v| k[v] = [] }
+        corporations.each do |corp|
+          reservations[corp.coordinates].unshift(corp)
```

Comment:
why are we unshifting here?

Reply:
It ensures home tokens are always first in the reservation list for cities like Erie in 1846 with reservations for multiple corps. Not essential though.